service/src/geopm/PlatformIO.hpp

```diff
@@ -248,6 +248,10 @@ namespace geopm
                                             int &server_pid,
                                             std::string &server_key) = 0;
             virtual void stop_batch_server(int server_pid) = 0;
+            /// @brief Query to see if the signal is valid.
+            ///
+            /// @param [in] name Signal to verify.
+            virtual bool signal_exists(const std::string &signal_name) const = 0;
```

Comment:
Although in principle this interface seems useful, we cannot simply append to the PlatformIO C++ interface.  This interface is exposed in C, Python, and through DBus in addition to the C++ implementation.  To add such a public interface to PlatformIO would require wrappers for all these other APIs.  

In addition, the implementation for this method is easily accomplished with an existing PlatformIO interface in all supported languages with something like:

```
    auto signal_names = platform_io().signal_names();
    bool signal_exists = (signal_names.find(signal_name) != signal_names.end());
```

similar code can be used for controls and the `control_names()` method.  I'm not in favor of adding an interface which contains logic that can be easily derived from another interface.